extensions/amp-fx-collection/0.1/providers/fx-provider.js

```diff
@@ -18,81 +18,128 @@ import {
   PositionObserverFidelity,
 } from '../../../../src/service/position-observer/position-observer-worker';
 import {Presets} from './amp-fx-presets';
+import {
+  ScrollToggleDispatch,
+  ScrollTogglePosition, // eslint-disable-line no-unused-vars
+  getScrollToggleFloatInOffset,
+  getScrollTogglePosition,
+  installScrollToggleStyles,
+  scrollToggleFloatIn,
+  userAsertValidScrollToggleElement,
+} from '../scroll-toggle';
 import {Services} from '../../../../src/services';
-import {assertDoesNotContainDisplay, setStyles} from '../../../../src/style';
-import {convertEasingKeyword, defaultDurationValues,
-  defaultEasingValues, defaultFlyInDistanceValues,
-  defaultMarginValues, installStyles, resolvePercentageToNumber}
-  from './amp-fx-presets-utils';
-import {getServiceForDoc} from '../../../../src/service';
+import {
+  assertDoesNotContainDisplay,
+  computedStyle,
+  setStyles,
+} from '../../../../src/style';
+import {
+  convertEasingKeyword,
+  defaultDurationValues,
+  defaultEasingValues,
+  defaultFlyInDistanceValues,
+  defaultMarginValues,
+  installStyles,
+  resolvePercentageToNumber,
+} from './amp-fx-presets-utils';
+import {
+  getServiceForDoc,
+  registerServiceBuilderForDoc,
+} from '../../../../src/service';
 import {
   installPositionObserverServiceForDoc,
 } from '../../../../src/service/position-observer/position-observer-impl';
 
+
 /**
- * Class that implements the various preset animation providers.
+ * @param {!../../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+ * @param {!Element} element The element to give a preset effect.
+ * @param {string} type
  */
-export class FxProvider {
```

Comment:
@cvializ @aghassemi This class was akin to a simple function and was simply keeping memory alive, simplified.